### ISSUE TITLE
Enforce `TFunction` type usage from `i18next`

### ISSUE THREAD

**Issue by tupe12334** [edvardchen/eslint-plugin-i18next#146]
**Description**

Currently, developers can define their own custom function types for translations, which can lead to inconsistent typing and runtime issues across the codebase.

To ensure consistency and maintain proper type safety, this rule should **enforce the use of the built-in `TFunction` type** from the `i18next` library instead of custom-defined types.

---

### ❌ Incorrect Code
```ts
type CustomTranslate = (key: string) => string;

const t: CustomTranslate = (key) => key;
```

### ✅ Correct Code
```ts
import { TFunction } from 'i18next';

const t: TFunction = (key) => key;
```

---

### Why this is needed
- Prevents fragmentation by ensuring all translations are typed in the same way.
- Improves type safety by relying on the well-maintained official i18next typings.
- Simplifies maintenance and onboarding for teams using i18next.

---

### Proposed Rule Name
`enforce-tfunction-type`

---

### Additional Notes
- The rule should provide an **auto-fix** where possible, replacing custom function types with `TFunction`.
- It should detect both **type aliases** and **interfaces** that replicate the `TFunction` signature.
- Should work well in projects using TypeScript.

---

### Example Rule Outline
```ts
module.exports = {
  meta: {
    type: 'problem',
    docs: {
      description: 'Enforce using the TFunction type from i18next instead of custom translation function types',
      category: 'Best Practices',
      recommended: false,
    },
    fixable: 'code',
    schema: [],
    messages: {
      useTFunction: 'Use the TFunction type from i18next instead of defining a custom function type.',
    },
  },
  create(context) {
    return {
      TSTypeAliasDeclaration(node) {
        if (
          node.typeAnnotation.type === 'TSFunctionType' &&
          node.typeAnnotation.parameters.length === 1 &&
          node.typeAnnotation.typeAnnotation?.type === 'TSStringKeyword'
        ) {
          context.report({
            node,
            messageId: 'useTFunction',
          });
        }
      },
    };
  },
};
```
